### PR TITLE
PG-1072: Fix major event trigger issues

### PR DESCRIPTION
**Pull request by dutow**
* Create table always checked the principal key and tried to create it, event when we didn't try to create a tde_heap table
* Alter table wasn't handled, and because of this changing a table to tde_heap access method didn't result in an encrypted table.
* defaut_table_access_method wasn't handled, and because of this, creating a table using that also resulted in a non encrypted table.

### CODE REVIEW

File: src/pg_tde_event_capture.c [hunk 1]
```diff
@@ -106,24 +107,69 @@ pg_tde_ddl_command_start_capture(PG_FUNCTION_ARGS)
...
+						(errmsg("failed to retrieve principal key. Create one using pg_tde_set_principal_key before using encrypted tables.")));
 
+			}
 		}
 
 	}
+	else if (IsA(parsetree, AlterTableStmt))
+	{
+		fprintf(stderr, "ALTEEEER\n");
+		fprintf(stderr, "ALTEEEER\n");
```
Comment (dAdAbird) [inline src/pg_tde_event_capture.c:137]: debug leftovers?

File: src/pg_tde_event_capture.c [hunk 2]
```diff
@@ -106,24 +107,69 @@ pg_tde_ddl_command_start_capture(PG_FUNCTION_ARGS)
...
+		Relation	rel = table_open(relationId, lockmode);
+		Oid tablespace_oid = rel->rd_locator.spcOid;
+		ListCell   *lcmd;
+		table_close(rel, lockmode);
+
+		foreach(lcmd, stmt->cmds)
+        	{
+			AlterTableCmd *cmd = (AlterTableCmd *) lfirst(lcmd);
+			if(cmd->subtype == AT_SetAccessMethod && strcmp(cmd->name, "tde_heap")==0) {
+		fprintf(stderr, "TO ENCRYPTED\n");
```
Comment (dAdAbird) [inline src/pg_tde_event_capture.c:151]: debug leftovers?

**Review by dAdAbird — APPROVED** [review]
LGTM overall just a couple of printf leftovers

File: src/pg_tde_event_capture.c (hunk 2 above)
Comment (dutow) [reply src/pg_tde_event_capture.c]: yes, fixed.

**Pull request merged**